Assets/_BForBoss/_Weapons/Scripts/Bullets/BulletBehaviour.ApplyDamage.cs

```diff
@@ -1,13 +1,18 @@
-using Perigon.Entities;
 using UnityEngine;
 
 namespace Perigon.Weapons
 {
+    public interface ILifeCycleBehaviour
```

Comment:
Should `ILifeCycleBehaviour` be within the `Perigon.Weapons` namespace? seems a bit off!

Reply:
Would it be better within `Perigon.Entities` so it matches the same namespace as `LifeCycleBehaviour` and `LifeCycle`?

Reply:
No, this has to be within weapons since its part of how the dependency works.

Anything that wants to use weapons must confirm to this interface to use it.

I can rename this to something else if need be.

Reply:
No, this has to be within weapons since its part of how the dependency works.

Anything that wants to use weapons must confirm to this interface to use it.

I can rename this to something else if need be.

Reply:
Yup please rename it then! Seems inconsistent IMO otherwise

Reply:
Let me know if there's any better naming.
IWeaponHolder
IWeaponUser